doc/community/developer-guide.md

```diff
@@ -137,9 +228,59 @@ def histogram(payload, table, column, bins, with_=None, conn=None):
 
 +++
 
-## Unit testing
+### Reading snippets
+
+JupySQL allows users to store snippets:
+
+```{code-cell} ipython3
+%sql sqlite-sqlalchemy
+```
+
+```{code-cell} ipython3
+%%sql --save fav_number
+SELECT * FROM foo WHERE bar = 42
+```
+
+These snippets help them break complex logic in multiple cells and automatically generate  CTEs. Now that we saved `fav_number` we can run `SELECT * FROM fav_number`, and JupySQL will automatically build the CTE:
+
+```{code-cell} ipython3
+%%sql
+SELECT * FROM fav_number WHERE bar = 42
+```
+
+In some scenarios, we want to allow users to use existing snippets for certain features. For example, we allow them to define a snippet and then plot the results using `%sqlplot`. If you're writing a feature that should support snippets, then you can use the `with_` argument in `raw_execute` and `execute`:
+
+```{code-cell} ipython3
+results = conn_sqlalchemy.raw_execute("SELECT * FROM fav_number", with_=["fav_number"])
+results.fetchall()
+```
+
+```{code-cell} ipython3
+results = conn_dbapi.raw_execute("SELECT * FROM fav_number", with_=["fav_number"])
+results.fetchall()
+```
+
+### `dialect`
+
+If you need to know the database dialect, you can access the `dialect` property in `SQLAlchemyConnection`s:
+
+```{code-cell} ipython3
+conn_sqlalchemy.dialect
+```
+
+Dialect in `DBAPIConnection` is only implemented for DuckDB, for all others, it currently returns `None`:
+
+```{code-cell} ipython3
+conn_dbapi.dialect is None
+```
+
+### `autocommit`
+
++++
```

Comment:
No details mentioned here

Reply:
deleted the section ( it isn't relevant)